File: backend/clients/qdrant.py
import os

from dotenv import load_dotenv
from qdrant_client import QdrantClient

load_dotenv()

QDRANT_URL = os.environ.get("QDRANT_URL")
QDRANT_API_KEY = os.environ.get("QDRANT_API_KEY")

if not all([QDRANT_URL, QDRANT_API_KEY]):
    raise RuntimeError(
        "Missing required env vars: QDRANT_URL, QDRANT_API_KEY. "
        "Please set them in your .env file."
    )

qdrant_client = QdrantClient(
    url=QDRANT_URL,
    api_key=QDRANT_API_KEY,
    timeout=60,
)

File: frontend/src/lib/core/rag/clients/qdrant.ts
import { QdrantClient } from "@qdrant/js-client-rest";

let client: QdrantClient | null = null;

export function getQdrantClient(): QdrantClient {
  const url = process.env.QDRANT_URL;
  const apiKey = process.env.QDRANT_API_KEY;
  if (!url || !apiKey) {
    throw new Error("QDRANT_URL and QDRANT_API_KEY must be configured");
  }
  if (!client) {
    client = new QdrantClient({
      url,
      apiKey,
      timeout: 120_000,
      checkCompatibility: false,
    });
  }
  return client;
}
